src/test/java/net/snowflake/client/jdbc/ResultSetMultiTimeZoneIT.java

```diff
@@ -317,6 +317,20 @@ public void testGetOldDate() throws SQLException {
     connection.close();
   }
 
+  @Test
+  public void testGetStringForDates() throws SQLException {
+    Connection connection = init();
+    Statement statement = connection.createStatement();
+    String expectedDate1 = "2020-08-01";
+    String expectedDate2 = "1920-11-11";
+    ResultSet rs = statement.executeQuery("SELECT '" + expectedDate1 + "'::DATE as D1");
+    rs.next();
+    assertEquals(expectedDate1, rs.getString(1));
+    rs = statement.executeQuery("SELECT '" + expectedDate2 + "'::DATE as D1");
+    rs.next();
+    assertEquals(expectedDate2, rs.getString(1));
+  }
+
```

Comment:
Just to double check, with the offset, we will get a different date? (I noticed we don't specify hour:minute:second now)

Reply:
Yes, it currently returns a different date when offset from certain timezones, which is wrong.